coderd/notifications/dispatch/webhook.go

```diff
@@ -0,0 +1,105 @@
+package dispatch
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+
+	"github.com/google/uuid"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+
+	"github.com/coder/coder/v2/coderd/notifications/types"
+	markdown "github.com/coder/coder/v2/coderd/render"
+	"github.com/coder/coder/v2/codersdk"
+)
+
+// WebhookHandler dispatches notification messages via an HTTP POST webhook.
+type WebhookHandler struct {
+	cfg codersdk.NotificationsWebhookConfig
+	log slog.Logger
+
+	cl *http.Client
+}
+
+// WebhookPayload describes the JSON payload to be delivered to the configured webhook endpoint.
+type WebhookPayload struct {
+	Version string               `json:"_version"`
+	MsgID   uuid.UUID            `json:"msg_id"`
+	Payload types.MessagePayload `json:"payload"`
+	Title   string               `json:"title"`
+	Body    string               `json:"body"`
+}
+
+func NewWebhookHandler(cfg codersdk.NotificationsWebhookConfig, log slog.Logger) *WebhookHandler {
+	return &WebhookHandler{cfg: cfg, log: log, cl: &http.Client{}}
```

Comment:
You may want to check how this handler behaves if the webhook endpoint is not responding, will it timeout eventually or hang